corehq/project_limits/models.py

```diff
@@ -113,6 +115,17 @@ def delete(self, *args, **kwargs):
         SystemLimit._get_global_limit.clear(self.key)
         SystemLimit._get_domain_specific_limit.clear(self.key, self.domain)
 
+    def _prevent_changing_from_global_to_domain_scope(self):
+        if self.domain:
+            try:
+                previous_domain = SystemLimit.objects.get(id=self.id).domain
+            except SystemLimit.DoesNotExist:
+                previous_domain = self.domain
```

Comment:
nit: This seems slightly easier to understand because it removes the need to mentally compute what will happen in the `if` statement below.
```suggestion
                return
```